app/controllers/manageiq/api/api_controller.rb

```diff
@@ -0,0 +1,172 @@
+        ApiController::Forbidden                 => :forbidden,
+        ApiController::BadRequestError           => :bad_request,
+        ApiController::NotFound                  => :not_found,
+        ApiController::UnsupportedMediaTypeError => :unsupported_media_type
+      }
+
+      #
+      # Support for REST API
+      #
+      include_concern 'Parameters'
+      include_concern 'Parser'
+      include_concern 'Manager'
+      include_concern 'Action'
+      include_concern 'Logger'
+      include_concern 'ErrorHandler'
+      include_concern 'Normalizer'
+      include_concern 'Renderer'
+      include_concern 'Results'
+
+      #
+      # Support for API Collections
+      #
+      include_concern 'Entrypoint'
+      include_concern 'Generic'
+
+      include_concern 'Accounts'
+      include_concern 'ArbitrationProfiles'
+      include_concern 'ArbitrationRules'
+      include_concern 'Authentication'
+      include_concern 'AutomationRequests'
+      include_concern 'Blueprints'
+      include_concern 'Categories'
+      include_concern 'CloudNetworks'
+      include_concern 'Conditions'
+      include_concern 'ContainerDeployments'
+      include_concern 'CustomAttributes'
+      include_concern 'Events'
+      include_concern 'Features'
+      include_concern 'Groups'
+      include_concern 'Hosts'
+      include_concern 'Instances'
+      include_concern 'Policies'
+      include_concern 'PolicyActions'
+      include_concern 'Providers'
+      include_concern 'ProvisionRequests'
+      include_concern "Rates"
+      include_concern "Reports"
+      include_concern 'Requests'
+      include_concern 'RequestTasks'
+      include_concern 'ResourceActions'
+      include_concern 'Roles'
+      include_concern 'ServiceDialogs'
+      include_concern 'ServiceOrders'
+      include_concern 'ServiceRequests'
+      include_concern 'Services'
+      include_concern 'ServiceTemplates'
+      include_concern 'Settings'
+      include_concern 'Software'
+      include_concern 'Tags'
+      include_concern 'TenantQuotas'
+      include_concern 'Tenants'
+      include_concern 'Users'
+      include_concern 'Vms'
+
+      #
+      # Api Controller Hooks
+      #
+      extend ErrorHandler::ClassMethods
+      respond_to :json
+      rescue_from_api_errors
+      prepend_before_action :require_api_user_or_token, :except => [:handle_options_request]
+
+      TAG_NAMESPACE = "/managed"
+
+      #
+      # Attributes used for identification
+      #
+      ID_ATTRS = %w(href id)
+
+      #
+      # To skip CSRF token verification as API clients would
+      # not have these. They would instead dealing with the /api/auth
+      # mechanism.
+      #
+      if Vmdb::Application.config.action_controller.allow_forgery_protection
+        skip_before_action :verify_authenticity_token, :only => [:show, :update, :destroy, :handle_options_request]
+      end
+
+      def base_config
+        ManageIQ::API::Settings.base
```

Comment:
sounds like one of the first to refactor then:  ManageIQ::API::SettingsController.